packages/block-library/src/paragraph/block.json

```diff
@@ -7,6 +7,7 @@
 	"description": "Start with the basic building block of all narrative.",
 	"keywords": [ "text" ],
 	"textdomain": "default",
+	"usesContext": [ "postId" ],
```

Comment:
I wonder whose responsibility is it to do that. In other words, the block other is not using the "postId" explicitly, so why add the "useContext" here. The "connection framework" should inject/filter this somehow.

Reply:
Do you mean that in the context of Core blocks or custom blocks or both?

Reply:
both

Reply:
OK! For now, I think that for core blocks this is fine. But you're right that for custom blocks we'll have to come up with a proper mechanism here. Are you all right with that?

Reply:
Yeah I think it's part of the definition of a "custom source" to say which change it needs to make to block registration. I think we should use it for both core and third-party blocks but yeah it's fine to start without it.

Reply:
For now, you could update `usesContext` and pass `postId` in PHP when processing `gutenberg_render_block_connections`. This way, it won't get unnecessarily wired for all Paragraph block instances.

Reply:
Even though this is the first implementation and custom blocks are not supported yet, I believe it'd be better to start creating mechanisms that would work for both if possible. Mainly because we already know we will want to support that in the future.

Reply:
Coming in a bit late here.

The current work on partial syncing (#54233) is using the store actions/selectors for attributes to inject/update values, but it might be difficult to pull this off if block context is being used for custom fields, as I believe it uses react context in JS.

While they're still proofs of concepts, the current work for custom fields/partial syncing does need to remain compatible, so it might be good to explore ways to read/write custom field values in the editor soon though so that we have a clear idea of the problems that need to be solved.